editors/code/src/installation/language_server.ts

```diff
@@ -0,0 +1,119 @@
+import { unwrapNotNil } from "ts-not-nil";
+import { spawnSync } from "child_process";
+import * as vscode from "vscode";
+import * as path from "path";
+import { strict as assert } from "assert";
+import { promises as fs } from "fs";
+
+import { BinarySource } from "./interfaces";
+import { fetchLatestArtifactMetadata } from "./fetch_latest_artifact_metadata";
+import { downloadFile } from "./download_file";
+
+export async function downloadLatestLanguageServer(
+    {file: artifactFileName, dir: installationDir, repo}: BinarySource.GithubRelease
+) {
+    const binaryMetadata = await fetchLatestArtifactMetadata(repo, artifactFileName);
+
+    const {
+        releaseName,
+        downloadUrl
+    } = unwrapNotNil(binaryMetadata, `Latest GitHub release lacks "${artifactFileName}" file`);
```

Comment:
I'd just `!` here, it seems unlikely that this fires. Hopefully, TS adds non-null assertions behind a flag one day...

Reply:
Do you mean runtime non-null assertions? I doubt they will ever do this or at least in close future. If you look at their [goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals), they are very reluctant to introducing too much custom syntax and runtime behavior that differs from `JavaScript` and if postfix `!` is not included into ECMAScript standard as a utility to do non-null assertions at runtime it will never be so in `TypeScript`.

Reply:
Though if you want to remove this sanity-check assert from here, I won't argue.